src/main/java/org/jenkinsci/plugins/workflow/cps/view/InterpolatedSecretsAction.java

```diff
@@ -0,0 +1,164 @@
+        if (interpolatedWarnings.isEmpty()) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    public boolean isInProgress() {
+        return run.isBuilding();
+    }
+
+    @Override
+    public void onAttached(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @Override
+    public void onLoad(Run<?, ?> run) {
+        this.run = run;
+    }
+
+    @ExportedBean
+    public static class InterpolatedWarnings {
+        final String stepName;
+        final List<String> interpolatedVariables;
+        final Run run;
+        final String nodeId;
+
+        private InterpolatedWarnings(@Nonnull String stepName, @Nonnull List<String> interpolatedVariables, @Nonnull Run run, @Nonnull String nodeId) {
+            this.stepName = stepName;
+            this.interpolatedVariables = interpolatedVariables;
+            this.run = run;
+            this.nodeId = nodeId;
+        }
+
+        @Exported
+        public String getStepSignature() {
+            StringBuilder sb = new StringBuilder();
+            Map<String, Object> stepArguments;
+            try {
+                stepArguments = getStepArguments(run, nodeId);
+            } catch (IllegalStateException e) {
+                return "Unable to construct " +  stepName + ": " + e.getMessage();
+            }
+
+            sb.append(stepName + "(");
+            Set<Map.Entry<String, Object>> entrySet = stepArguments.entrySet();
+            if (!entrySet.isEmpty()) {
+                // Give some sort of order to the step arguments
+                TreeSet<String> sortedArgs = new TreeSet<>();
+                for (Map.Entry<String, Object> argEntry : stepArguments.entrySet()) {
+                    Object value = argEntry.getValue();
+                    String valueString = String.valueOf(value);
+                    if (value instanceof ArgumentsAction.NotStoredReason) {
+                        switch ((ArgumentsAction.NotStoredReason) value) {
+                            case OVERSIZE_VALUE:
+                                valueString = "argument omitted due to length";
+                                break;
+                            case UNSERIALIZABLE:
+                                valueString = "unable to serialize argument";
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                    sortedArgs.add(argEntry.getKey() + ": " + valueString);
+                }
+                Iterator<String> it = sortedArgs.iterator();
+                sb.append(it.next());
+                while (it.hasNext()) {
+                    sb.append(", " + it.next());
+                }
+            }
+            sb.append(")");
+            return sb.toString();
+        }
+
+        @Nonnull
+        private Map<String, Object> getStepArguments(Run run, String nodeId) throws IllegalStateException {
```

Comment:
There were so many `@CheckForNull` conditions that bloated the code that I decided to just put it in its own method.